src/TableView.jl

```diff
@@ -1,119 +1,158 @@
 module TableView
-
-using WebIO
-using JSExpr
-using JuliaDB
-using DataValues
-
-import JuliaDB: DNDSparse, DNextTable, NextTable
-
-function JuliaDB.subtable(t::DNextTable, r)
-    table(collect(rows(t)[r]), pkey=t.pkey)
-end
-
-showna(xs) = xs
-function showna(xs::AbstractArray{T}) where {T<:DataValue}
-    map(xs) do x
-        isnull(x) ? "NA" : get(x)
+using Tables, TableTraits, IteratorInterfaceExtensions
+using WebIO, JSExpr, JSON, Dates
+using Observables: @map
+
+function showtable(x; dark = false)
+    if Tables.istable(typeof(x))
+        return _showtable(x, dark)
+    elseif TableTraits.isiterabletable(x)
+        it = IteratorInterfaceExtensions.getiterator(x)
+        return _showtable(Tables.DataValueUnwrapper(it), dark)
     end
+    throw(ArgumentError("Argument is not a table."))
 end
 
-function showna(xs::Columns)
-    rows(map(showna, columns(xs)))
+function _showtable(table, dark)
+    length(Tables.rows(table)) > 10_000 ? _showtable_async(table, dark) :
```

Comment:
This should check ``IteratorSize`` before calling ``length``. Any [Query.jl](https://github.com/queryverse/Query.jl) query that includes a ``@filter`` will have ``IteratorSize(q)==SizeUnknown()``, and so a sink needs to be able to handle sources with unknown length.